File: 290-WordPattern/290-WordPattern.java
// Last updated: 09/07/2026, 10:08:33
class Solution {
    public boolean wordPattern(String pattern, String s) {
        HashMap<Character, String> c_w = new HashMap<>();
        HashMap<String, Character> w_s = new HashMap<>();
        String[] words = s.split(" ");
        if(pattern.length()!=words.length){
            return false;
        }
        for(int i = 0;i<words.length;i++){
            char ch = pattern.charAt(i);
            String st = words[i];
            if(c_w.containsKey(ch) && !c_w.get(ch).equals(st)){
                return false;
            }
            if(w_s.containsKey(st) && !w_s.get(st).equals(ch)){
                return false;
            }
            c_w.put(ch,st);
            w_s.put(st,ch);

        }
        return true;
    }
}

File: 290-WordPattern/290-WordPattern.py
# Last updated: 09/07/2026, 10:08:37
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        words = s.split()
        if len(words) != len(pattern):
            return False
        
        h = {}
        m = {}
        for p,w in zip(pattern,words):
            if p in h:
                if h[p] != w:
                    return False
            else:
                h[p] = w

            if w in m:
                if m[w] != p:
                    return False    

            else:
                m[w] = p        

        return True     